src/hotspot/share/compiler/methodMatcher.cpp

```diff
@@ -269,6 +270,19 @@ void MethodMatcher::parse_method_pattern(char*& line, const char*& error_msg, Me
     c_match = check_mode(class_name, error_msg);
     m_match = check_mode(method_name, error_msg);
 
+    // Over-consumption
+    // method_name points to an option type or option name because the method name is not specified by users.
+    // In very rare case, the method name happens to be same as option type/name, so look ahead to make sure
+    // it doesn't show up again.
+    if ((OptionType::Unknown != parse_option_type(method_name) ||
+        CompileCommand::Unknown != CompilerOracle::string_to_option(method_name)) &&
+        strstr(line + bytes_read, method_name) == NULL) {
+      error_msg = "not specify any method pattern";
+      m_match = MethodMatcher::Unknown;
+      bytes_read -= strlen(method_name);
```

Comment:
Windows build fails on this with "conversion from 'size_t' to 'int'"

Reply:
Thanks!  I delete this statement because bytes_read will not be used. 
The function exits early because of m_match = MethodMatcher::Unknown.